native/src/components/TtsContainer.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { createContext, ReactElement, useCallback, useEffect, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { AppState } from 'react-native'
+import Tts from 'react-native-tts'
+
+import buildConfig from '../constants/buildConfig'
+import { AppContext } from '../contexts/AppContextProvider'
+import { reportError } from '../utils/sentry'
+import TtsPlayer from './TtsPlayer'
+
+const unsupportedLanguagesForTts = ['fa', 'ka', 'kmr']
+
+export const isTtsActive = (content: string[] | null, languageCode: string): boolean =>
+  Array.isArray(content) &&
+  content.length > 0 &&
+  Boolean(buildConfig().featureFlags.tts) &&
+  !unsupportedLanguagesForTts.includes(languageCode)
+
+export type TtsContextType = {
+  visible: boolean
+  setVisible: (visible: boolean) => void
+  sentences: string[] | null
+  setSentences: (sentences: string[]) => void
+}
+
+export const ttsContext = createContext<TtsContextType>({
+  visible: false,
+  setVisible: () => undefined,
+  sentences: [],
+  setSentences: () => undefined,
+})
+
+type TtsContainerProps = {
+  children: ReactElement
+}
+
+const TtsContainer = ({ children }: TtsContainerProps): ReactElement => {
+  const { languageCode } = React.useContext(AppContext)
+  const { t } = useTranslation('layout')
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [isExpanded, setIsExpanded] = useState(false)
```

Comment:
:x: I am still very sure that this `expanded` state is redundant if the other suggestions are applied. In my opinion, there should only be three states necessary: enabled/disabled (if disabled in the build config, no actual useState necessary), visible/hidden, playing/not playing. If it is playing, it should automatically expand (as it is now implemented). If it is paused, it should be collapsed. If the value of the useState `isPlaying` does not change anymore after every sentence/forward/backward, this property will therefore not be needed anymore.